core-web/libs/edit-content/src/lib/dot-edit-content.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Observable } from 'rxjs';
+
+import { HttpClient } from '@angular/common/http';
+import { Injectable, inject } from '@angular/core';
+
+import { map, pluck } from 'rxjs/operators';
+
+import { DotContentTypeService, DotWorkflowActionsFireService } from '@dotcms/data-access';
+import { DotCMSContentType } from '@dotcms/dotcms-models';
+
+import { DotForm } from './interfaces/dot-form.interface';
+
+@Injectable()
+export class DotEditContentService {
+    private readonly dotContentTypeService = inject(DotContentTypeService);
+    private readonly dotWorkflowActionsFireService = inject(DotWorkflowActionsFireService);
+    private readonly http = inject(HttpClient);
+
+    getContentType(id: string): Observable<DotCMSContentType> {
+        return this.http.get(`/api/v1/content/${id}`).pipe(pluck('entity'));
+    }
+
+    getContent(idOrVar: string): Observable<DotForm[]> {
+        return this.dotContentTypeService.getContentType(idOrVar).pipe(
+            map((content: DotCMSContentType) => {
+                const mappedData = content.layout.map((row) => {
+                    const columns = row.columns?.map((column) => {
+                        const fields = column.fields.map((field) => {
+                            return {
+                                id: field.id,
+                                type: field.fieldType,
+                                hint: field.hint,
+                                label: field.name,
+                                required: field.required,
+                                regexCheck: field.regexCheck,
+                                // url: field.url,
+                                variable: field.variable
+                            };
+                        });
+
+                        return { fields };
+                    });
+
+                    return { row: { columns } };
+                });
+
+                return mappedData as DotForm[];
+            })
+        );
```

Comment:
Maybe we can rewrite this. It's just a suggestion. You can skip this if you want.

```javascript
map(({ layout }: DotCMSContentType) =>
    layout.map(({ columns = [] }) => ({
        row: {
            columns: columns.map(({ fields }) => ({
                fields: fields.map(
                    ({ id, fieldType, hint, fieldTypeLabel, required, regexCheck, variable }) => ({
                        type: fieldType,
                        label: fieldTypeLabel,
                        id,
                        hint,
                        required,
                        regexCheck,
                        variable
                    })
                )
            }))
        }
    }))
);
```

Reply:
We had a meeting and realized that this is not needed at all